src/sentry/issues/escalating_group_forecast.py

```diff
@@ -47,15 +49,27 @@ def save(self) -> None:
             ttl=timedelta(GROUP_FORECAST_TTL),
         )
 
+    @classmethod
+    def _should_fetch_escalating(cls, group_id: int, project_id: int) -> bool:
+        group = Group.objects.get(id=group_id)
+        organization = Project.objects.get(id=project_id).organization
+        return group.issue_type.should_detect_escalation(organization)
```

Comment:
This will make 3 queries to get the organization. Instead, you should be able to do something like

```
Organization.objects.get(projects_set__group__set__id=group_id)
```

I don't know the exact name of the relations, but you can look them up on the models. This will fetch the org using just one query.